### PR TITLE
chore(opentelemetry): add trace state and trace flags to SpanContext

### PR DESCRIPTION
**Pull request by mabdinur**
Currently Open Telemetry spans generated by the ddtrace library do not provide an interface for accessing the trace state and trace flags. This does not align with the spec. 

This PR ensures the span context returned by Span.get_span_context() contains values for trace state and trace flags. 

OpenTelemetry SpanContext API: https://github.com/open-telemetry/opentelemetry-python/blob/v1.17.0/opentelemetry-api/src/opentelemetry/trace/span.py#L406 
OpenTelmetry SpanContext docs: https://opentelemetry.io/docs/reference/specification/trace/api/#spancontext

ddtrace openTelemetry support is currently unannounced. So this change does not require a release note.

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/contributing.html#Release-Note-Guidelines) are followed.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Author is aware of the performance implications of this PR as reported in the benchmarks PR comment.

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer is aware of, and discussed the performance implications of this PR as reported in the benchmarks PR comment.

### CODE REVIEW

File: ddtrace/_opentelemetry/span.py [hunk 1]
```diff
@@ -92,8 +94,15 @@ def get_span_context(self):
         """Returns an Open Telemetry SpanContext"""
         # Returns a SpanContext with the current trace id, span id and is_remote flag.
         # This is the bare minimum to get current context.
-        # TODO: Support propagating TraceState and TraceFlag
-        return SpanContext(self._ddspan.trace_id, self._ddspan.span_id, False)
+        tf = TraceFlags.DEFAULT
```
Comment (mabdinur) [inline ddtrace/_opentelemetry/span.py:97]: There are two trace flags: default and sampled

File: tests/opentelemetry/test_span.py [hunk 2]
```diff
@@ -113,3 +116,35 @@ def test_otel_span_exception_handling(oteltracer):
...
+    span_context = otelspan.get_span_context()
+    assert span_context.trace_id == otelspan._ddspan.trace_id
+    assert span_context.span_id == otelspan._ddspan.span_id
+    # An ddtrace._opentelemetry.Span can never be remote.
+    # opentelemetry.trace.NonRecordingSpan is used to represent a "remote span".
+    assert span_context.is_remote is False
+    # By default ddtrace set sampled=True for all spans
+    assert span_context.trace_flags == TraceFlags.SAMPLED
+    # Default tracestate values set on all Datadog Spans (Note - these values may change)
+    assert span_context.trace_state.to_header() == "dd=s:1;t.dm:-0"
```
[the remark this answers is not in the document]
Comment (mabdinur) [reply tests/opentelemetry/test_span.py:133]: This attribute is defined in the otel api: https://github.com/open-telemetry/opentelemetry-python/blob/v1.17.0/opentelemetry-api/src/opentelemetry/trace/span.py#L428. We can't change it but we could update ddtrace.Context._tracestate to be consistent with this value.

File: tests/opentelemetry/test_span.py [hunk 3]
```diff
@@ -113,3 +116,35 @@ def test_otel_span_exception_handling(oteltracer):
...
+
+    span_context = otelspan.get_span_context()
+    assert span_context.trace_id == otelspan._ddspan.trace_id
+    assert span_context.span_id == otelspan._ddspan.span_id
+    # A ddtrace._opentelemetry.Span can never be remote.
+    # opentelemetry.trace.NonRecordingSpan is used to represent a "remote span".
+    assert span_context.is_remote is False
+    # By default ddtrace set sampled=True for all spans
+    assert span_context.trace_flags == TraceFlags.SAMPLED
+    # Default tracestate values set on all Datadog Spans (Note - these values may change)
```
Comment (mabdinur) [inline tests/opentelemetry/test_span.py:132]:
```suggestion
    # Default tracestate values set on all Datadog Spans
```

File: tests/opentelemetry/test_span.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ZStriker19) [reply tests/opentelemetry/test_span.py:133]: ah how strange... I feel like we should update it for our tracer where we can? But, I'm also ok with not if you don't think we should.

[the remark this answers is not in the document]
Comment (mabdinur) [reply tests/opentelemetry/test_span.py:133]: I think it's fine, if we get time we can do it as a follow up. `opentelemetry.trace.span.SpanContext` and `ddtrace.Context` are significantly different. Aligning on the name of one property is not the biggest deal.

**Review by ZStriker19 — APPROVED** [review]

**Pull request merged**